### PR TITLE
Rebase dashboard

### PR DESCRIPTION
**Pull request by 21Chani**
This pull request will add user-dashboard into app

### CODE REVIEW

File: apps/cave/components/UserDashboard/Summary/Staking/LiquidStakingSnapshot.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  const networkId = useCurrentSupportedNetworkId()
+
+  const airdropClaimableAmount = getAirdropQ4ClaimableAmount(address)
+  const airdropAmount = airdropClaimableAmount || 0
+  const { data: claimed } = useQuery(['AirdropClaimContract', networkId], async () => {
+    const airdrop = new AirdropClaimContract(concaveProvider(networkId), 'Q4')
+    return await airdrop.claimed(address)
+  })
+
+  const airdropTotal = 510691.11 //TODO
```
Comment (Schrammel) [inline apps/cave/components/UserDashboard/Summary/Staking/LiquidStakingSnapshot.tsx:40]: @21Chani ?

Comment (psgb-r3763) [reply apps/cave/components/UserDashboard/Summary/Staking/LiquidStakingSnapshot.tsx:40]: this is left over from when it was hardcoded, unsure where to get the total token distribution at the time. ideally we can pull this from somewhere reliable, maybe onchain?

File: apps/cave/components/UserDashboard/Summary/TxHistory/DashItem.type.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export type DashItem = {
+  amountIn: number | string
+  amountOut: number | string
+  currencyIn: string
+  currencyOut: string
+  category: string
+  eventType: string
```
Comment (psgb-r3763) [inline apps/cave/components/UserDashboard/Summary/TxHistory/DashItem.type.ts:7]: `eventType` should become `txType` soon. its a better name

File: apps/cave/components/UserDashboard/Summary/TxHistory/TxHistory.tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    enableSorting: false,
+  }),
+  columnHelper.accessor('category', {
+    id: 'category',
+    header: () => 'Category',
+    cell: (info) => info.getValue(),
+    enableMultiSort: true,
+  }),
+  columnHelper.accessor('eventType', {
+    id: 'eventType',
```
Comment (psgb-r3763) [inline apps/cave/components/UserDashboard/Summary/TxHistory/TxHistory.tsx:131]: eventType should become txType soon. its a better name

File: apps/cave/components/UserDashboard/Summary/Bonding/BondingSnapshot.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import { DataTable } from 'components/UserDashboard/DataTable'
+import { SnapshotLineChart } from 'components/UserDashboard/SnapshotLineChart'
+import { SnapshotTextCard } from 'components/UserDashboard/SnapshotTextCard'
+import { useCNVPrice } from 'hooks/useCNVPrice'
+import { useState } from 'react'
+import { DataTableCard } from '../../DataTableCard'
+import { useUserBondState } from '../../hooks/useUserBondState'
+import { SnapshotCard } from '../../SnapshotCard'
+import { SnapshotText } from '../../SnapshotText'
+import { bondchartdata } from '../dummyChartData'
```
Comment (psgb-r3763) [inline apps/cave/components/UserDashboard/Summary/Bonding/BondingSnapshot.tsx:12]: implementing with dummy data?

File: apps/cave/components/UserDashboard/Summary/Staking/LiquidStakingSnapshot.tsx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+import { concaveProvider } from 'lib/providers'
+import { useState } from 'react'
+import { useQuery } from 'react-query'
+import { useAccount } from 'wagmi'
+import { UserPositionCard } from '../../../StakingPositions/LockPosition/Card/UserPositionCard'
+import { DataTable } from '../../DataTable'
+import { DataTableCard } from '../../DataTableCard'
+import { SnapshotCard } from '../../SnapshotCard'
+import { SnapshotText } from '../../SnapshotText'
+import { stakechartdata } from '../dummyChartData'
```
Comment (psgb-r3763) [inline apps/cave/components/UserDashboard/Summary/Staking/LiquidStakingSnapshot.tsx:22]: implmenting with dummy data?

Comment (Schrammel) [reply apps/cave/components/UserDashboard/Summary/Staking/LiquidStakingSnapshot.tsx:25]: :(

File: apps/cave/components/UserDashboard/Summary/Staking/LiquidStakingSnapshot.tsx (hunk 1 above)
Comment (Schrammel) [reply apps/cave/components/UserDashboard/Summary/Staking/LiquidStakingSnapshot.tsx]: good @psgb-r3763 i did it

Comment (psgb-r3763) [reply apps/cave/components/UserDashboard/Summary/Staking/LiquidStakingSnapshot.tsx]: new version should be good!

File: apps/cave/components/UserDashboard/Summary/Staking/LiquidStakingSnapshot.tsx [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+const useAirdropOverview = (address:string, chainId) => {
+  return useQuery(['AirdropClaimOverview', chainId], async () => {
+    const provider = concaveProvider(chainId)
+    const seasons = [{
+        claimed: await new AirdropClaimContract(provider, 'special').claimed(address),
+        amount: getAirdropSpecialClaimableAmount(address) || 0
+      },{
+        claimed: await new AirdropClaimContract(provider, 'Q4').claimed(address),
```
Comment (greg-schrammel) [inline apps/cave/components/UserDashboard/Summary/Staking/LiquidStakingSnapshot.tsx:34]: I'm not sure did you test if awaiting one after the other get's bundled into a single multicall or maybe we should do it like
```ts
// start both promises without blocking 
const claimedSpecial = new AirdropClaimContract(provider, 'special').claimed(address)
const claimedQ4 = new AirdropClaimContract(provider, 'Q4').claimed(address) 
// then await their response to return
const seasons = [{
        claimed: await claimedSpecial,
        amount: getAirdropSpecialClaimableAmount(address) || 0
      },{
        claimed: await claimedQ4,
```

Comment (Schrammel) [reply apps/cave/components/UserDashboard/Summary/Staking/LiquidStakingSnapshot.tsx:76]: are you right

File: apps/cave/components/LiquidityPoolPositions/hooks/usePositionsState.tsx [hunk 7]
```diff
@@ -6,7 +6,7 @@ import { useState } from 'react'
 import { useQuery } from 'react-query'
 import { useAccount } from 'wagmi'
 
-export const usePositionsState = () => {
+export const usePositionsState = (initialView?: 'user' | 'all' | undefined) => {
```
Comment (Schrammel) [inline apps/cave/components/LiquidityPoolPositions/hooks/usePositionsState.tsx:9]:
```suggestion
export const usePositionsState = (initialView?: 'user' | 'all' ) => {
```

**Review by Schrammel — APPROVED** [review]

**Pull request merged**